cjs/interface/event.js

```diff
@@ -36,8 +38,14 @@ const GlobalEvent = typeof Event === 'function' ?
     preventDefault() { this.defaultPrevented = true; }
 
     // TODO: what do these do in native NodeJS Event ?
-    stopPropagation() {}
-    stopImmediatePropagation() {}
+    stopPropagation() {
+      this.cancelBubble = true;
+    }
+    
+    stopImmediatePropagation() {
+      this.stopPropagation();
+      this._stopImmediatePropagationFlag = true;
+    }
```

Comment:
@WebReflection Node 16.5 has issues here:
`stopImmediatePropagation` should set both "stop propagation" and "stop immediate propagation" flags, but Node don't do that and only set their inner [kStop] value (specs https://dom.spec.whatwg.org/#dom-event-stopimmediatepropagation VS implementation of Event in https://github.com/nodejs/node/blob/v16.5.0/lib/internal/event_target.js).

- We have no way of knowing [kStop] as currently they don't have getter for it.

So with that I don't see any way for DOMEventTarget to prevent parent nodes listeners to be called,
unless maybe by extending node's Event with something like `DOMEvent`?

Reply:
we are extending the base class though, why wouldn't this work in our `DOMEventTarget` ?

```js
stopImmediatePropagation() {
  super.stopPropagation();
  super.stopImmediatePropagation();
}
```

Reply:
yes ... but ... doesn't that work on Event? not sure I am following

Reply:
So with node 12 and 14 everything works because it relies fully on linkedom implementations.

Node 15+ have Event in that case we use Node's versions which come with an incomplete `stopImmediatePropagation` (they don't have bubbling so for them it probably makes sense).

In the same way we had to extend EventTarget with DOMEventTarget I think it makes sense to extend Event too,
to add what is needed.

This is what I did in the last update, which ensure it works for all versions of node.

Reply:
> they don't have bubbling so for them it probably makes sense

well ... not really, `stopImmediatePropagation` means don't execute any other listener, bubbling a part ... but I don't care patching the world here, just trying not to break what was working before ;-)